pkg/resource/template/resource.go

```diff
@@ -155,16 +155,31 @@ func (t *TemplateResource) createStageFile() error {
 	}
 
 	if err = tmpl.Execute(temp, nil); err != nil {
-		temp.Close()
-		os.Remove(temp.Name())
+		// The key error to return is the failure to execute.
+		// to preserve that error ignore the errors in close and clean
+		temp.Close()           // nolint:errcheck
+		os.Remove(temp.Name()) // nolint:errcheck
 		return err
 	}
-	defer temp.Close()
+	defer func() {
+		if e := temp.Close(); e != nil {
```

Comment:
Do we really need to handle an error on Close().  Doesn't feel like there's a whole lot we can do for that one, and not one that should be propagated up.

Reply:
Feels like the logic is going awry.  I think we should be doing one of the following:
- tracing any error and carrying on regardless
- return the error up the stack and make sure we tidy up correctly (in this case, I think we should be deleting the file if we are returning an error)

Reply:
let's just log the error and not return it to the caller...it isn't adding much value.